returnn/torch/distributed.py

```diff
@@ -23,19 +23,31 @@ class DistributedContext:
     """
 
     def __init__(self, options: Dict[str, Any]):
-        import torch.distributed as dist
-
         self._opts = CollectionReadCheckCovered(options)
 
-        # when no backend is specified, both gloo and nccl backends will be created
-        # the gloo backend will be used for collectives with CPU tensors and
-        # the nccl backend will be used for collectives with CUDA tensors
-        dist.init_process_group(backend=self._opts.get("backend", None))
+        # Subprocesses have issues initializing torch.distributed process groups.
+        #
+        # We therefore pass rank/size information of the process group via an env
+        # variable that is automatically inherited in any created subprocess.
+        env_var_name = "_RETURNN_TORCH_DISTRIBUTED_INIT_INFO"
+        prev_init_info = os.environ.get(env_var_name)
+        if prev_init_info:
+            self.prev_init_info = ast.literal_eval(prev_init_info)
+            self._rank = self.prev_init_info["rank"]
+            self._size = self.prev_init_info["size"]
+        else:
+            import torch.distributed as dist
+
+            # when no backend is specified, both gloo and nccl backends will be created
+            # the gloo backend will be used for collectives with CPU tensors and
+            # the nccl backend will be used for collectives with CUDA tensors
+            dist.init_process_group(backend=self._opts.get("backend", None))
+            self._rank = dist.get_rank()
+            self._size = dist.get_world_size()
+            os.environ[env_var_name] = str({"rank": self._rank, "size": self._size})
```

Comment:
```suggestion
            os.environ[env_var_name] = repr({"rank": self._rank, "size": self._size})
```
While it is the same for this specific case, it just makes more sense, and there are other cases where `str` would be wrong, and `repr` is what you actually want here.